File: frontend/src/Habit.ts
export type Habit = {
    id: string,
    name: string,
    description: string,
    dailySaving: number,
    startTime: Date,
    lastTimeCollected: Date,
    endTime: Date,
    statusOpen: boolean
}
export type NewHabit = {
    name: string,
    description: string,
    dailySaving: number
}

File: backend/src/main/java/de/neuefische/backend/habit/Habit.java
package de.neuefische.backend.habit;

import org.springframework.data.annotation.Id;

import java.time.OffsetDateTime;

public record Habit(
        @Id
        String id,
        String name,
        String description,
        double dailySaving,
        OffsetDateTime startTime,
        OffsetDateTime lastTimeCollected,
        OffsetDateTime endTime,
        boolean statusOpen

) {
}
